smdebug_rulesconfig/debugger_rules/_ruleGroups.py

```diff
@@ -1,29 +1,30 @@
-from enum import Enum
-
-from ._utils import _get_config_for_group
-
-
-class MXNET(Enum):
-    SIMPLE = ["VanishingGradient", "LossNotDecreasing", "WeightUpdateRatio"]
-    ALL = []
-
-
-class TENSORFLOW(Enum):
-    SIMPLE = ["VanishingGradient", "LossNotDecreasing", "WeightUpdateRatio"]
-    ALL = []
-
-
-class PYTORCH(Enum):
-    SIMPLE = ["VanishingGradient", "LossNotDecreasing", "WeightUpdateRatio"]
-    ALL = []
-
-
-class XGBOOST(Enum):
-    SIMPLE = ["TreeDepth", "ClassImbalance"]
-    ALL = []
-
-
-def get_rule_groups(ruleEnum):
-    ruleEnumVal = ruleEnum.value
-    rules_config = _get_config_for_group(ruleEnumVal)
-    return rules_config
+# set of rules that are expected to work for all supported frameworks
+# Supported Frameworks: Mxnet, Pytorch, Tensorflow, Xgboost
+UNIVERSAL_RULES = {
+    "AllZero",
+    "ClassImbalance",
+    "LossNotDecreasing",
+    "Overfit",
+    "Overtraining",
+    "SimilarAcrossRuns",
+    "StalledTrainingRule",
+    "TensorVariance",
+    "UnchangedTensor",
+}
+
+# set of rules that are expected to work for only for supported deep learning frameworks
+# Supported Deep Learning Frameworks: Mxnet, Pytorch, Tensorflow
+DEEP_LEARNING_RULES = {
+    "DeadRelu",
+    "ExplodingTensor",
+    "PoorWeightInitialization",
+    "SaturatedActivation",
+    "VanishingGradient",
+    "WeightUpdateRatio",
+}
+
+# Rules intended to be used as part of a DL Application
+DEEP_LEARNING_APPLICATION_RULES = {"CheckInputImages", "NLPSequenceRatio"}
+
+# Rules only compatible with XGBOOST
+XGBOOST_RULES = {"Confusion", "FeatureImportanceOverweight", "TreeDepth"}
```

Comment:
Confusion rule should also work for deep learning based classification models.

Reply:
Moved it.